### PR TITLE
massive changes...

### PR DESCRIPTION
**Pull request by gdementen**
Ahem. This is the massive PR you have been waiting for :)

You might want to review commits individually.

Sorry for that *and* for the massive commits within the PR. Shouldn't have done that much development without commiting, but it is too late now to do it right.

### CODE REVIEW

File: larray_editor/editor.py [hunk 1]
```diff
@@ -503,11 +503,12 @@ def update_title(self):
         self.setWindowTitle(' - '.join(title))
 
     def set_current_array(self, array, name):
-        if array is not self.current_array:
-            self.current_array = array
-            self.arraywidget.set_data(array)
-            self.current_array_name = name
-            self.update_title()
+        # we should NOT check that "array is not self.current_array" because this method is also called to
+        # refresh the widget value because of an inplace setitem
```
Comment (alixdamman) [inline larray_editor/editor.py]: OK but `set_current_array` is also called by `view_expr` which passes '<expr>' has name argument and overwrite the real name of the array.

File: larray_editor/utils.py [hunk 2]
```diff
@@ -390,3 +405,94 @@ def __getitem__(self, key):
 
     def __len__(self):
         return self.length
+
+
+def replace_inf(value):
+    """Replace -inf/+inf in array with respectively min(array_without_inf)/max(array_without_inf).
```
Comment (alixdamman) [inline larray_editor/utils.py]: I guess this is a function dedicated to be used internally, users should not know about it. Right?

Comment (gdementen) [reply larray_editor/utils.py]: indeed, why do you ask?

Comment (alixdamman) [reply larray_editor/utils.py]: It is fully documented

Comment (gdementen) [reply larray_editor/utils.py]: it does not hurt to document internal functions, right? :)

File: larray_editor/editor.py (hunk 1 above)
Comment (gdementen) [reply larray_editor/editor.py]: view_expr uses '<expr>' and this is perfectly fine since it is no longer the same array

**Comment by gdementen** [comment]
the build is failing because I rely on larray 0.26 syntax (to avoid creating warnings) and since it installs larray from conda-forge, it is still using version 0.25.2. I guess we should use a git checkout (or use pip to install from git) to avoid this kind of issue. This issue will go away if we manage to remerge larray-editor in larray, but in the meantime, what do we do?

**Comment by alixdamman** [comment]
merge

File: larray_editor/utils.py (hunk 2 above)
Comment (alixdamman) [reply larray_editor/utils.py]: right

**Pull request merged**